src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int size = 0;
+    private final int defaultSize = 10;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[defaultSize];
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        indexExeption(index);
+        resize();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexExeption(index);
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexExeption(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexExeption(index);
+        final T removedObject = elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        size--;
+        return removedObject;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (elements[i] != null && elements[i].equals(element) || element == elements[i]) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("No such element in array");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void resize() {
+        if (elements.length == size) {
+            T[] increasedObjectsArray = (T[]) new Object[(int) (elements.length * 1.5)];
```

Comment:
```suggestion
            T[] resizedElements = (T[]) new Object[(int) (elements.length * 1.5)];
```
Try to not use word `object` in variable names